app/src/main/java/com/lucas/medicaltools/service/MainRepository.kt

```diff
@@ -0,0 +1,47 @@
+package com.lucas.medicaltools.service
+
+import android.view.View
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.MainViewState
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.schedulers.Schedulers
+import retrofit2.Response
+import java.util.*
+object MainRepository {
```

Comment:
Make this a singleton ;)

Reply:
Ensure your repo is available in your presenter to use with your intents ;)

Reply:
In regards of a singleton, please see the attached comment.
![Uploading image.png…]()

https://www.geeksforgeeks.org/singleton-class-in-kotlin/#:~:text=Singleton%20Class%20in%20Kotlin%20is,be%20created%20and%20used%20everywhere.